src/authenticated/member/MemberProfilePage.tsx

```diff
@@ -0,0 +1,53 @@
+import gql from 'graphql-tag';
+import { graphql } from 'react-apollo';
+import MemberProfile, { MemberProfileFragment } from './MemberProfile';
+import {
+  getUser,
+  getUserVariables,
+  getUser_user
+} from './__generated__/getUser';
+
+const userQuery = gql`
+  query getUser($id: ID) {
+    user: User(id: $id) {
+      ...MemberProfileFragment
+    }
+  }
+  ${MemberProfileFragment}
+`;
+
+type MemberProfileOuterProps = {
+  match: {
+    params: {
+      id: string;
+    };
+  };
+};
+type ComponentProps = {
+  user?: getUser_user;
+  editing: boolean;
+  onFormDataChange: (value: any) => void;
+};
+
+const MemberProfilePage = graphql<
```

Comment:
Ooph - those typings are...verbose. I understand why we use recompose and the graphql function, but this seems like a broader architectural concern about the overuse of the "withAuthSession" HoC / helper in so many of the components. the <Query> component + render function seems like the recommended pattern if we don't use recompose. Perhaps @wdoug your concerns about the complex generics of the graphql function can be alleviated if we switch to Query component as much as possible. See typing for the Query component [here](https://www.apollographql.com/docs/react/recipes/static-typing.html#typing-components).